From fb7cd1d84f7d62535cc8b99f56db59e71e8d7cbc Mon Sep 17 00:00:00 2001
From: Nikhil R Deshpande <ndeshpande@vmware.com>
Date: Wed, 12 Apr 2017 19:05:17 -0700
Subject: [PATCH 7/7] Look for alternate WindowsKit lib dir on VC140 env

In some Windows build environments with a version of Windows Kit and
VC 14.0 compiler, the SDK lib dir is observed to be different. E.g.
sdk_dir/lib/winv6.3/um instead of sdk_dir/lib/win8/um.
This patch attempts to try looking for either of the paths before
giving up.
---
 setup.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/setup.py b/setup.py
index 590d97e..4112169 100644
--- a/setup.py
+++ b/setup.py
@@ -793,6 +793,11 @@ class my_build_ext(build_ext):
                 assert os.path.isdir(extra), extra
         else:
             extra = os.path.join(sdk_dir, 'lib', 'win8', 'um')
+            # If above sdk lib dir doesn't exist, then try another commonly
+            # seen path (there's an assert below that to catch even the
+            # alternative is missing).
+            if not os.path.isdir(extra):
+                extra = os.path.join(sdk_dir, 'lib', 'winv6.3', 'um')
             if is_64bit:
                 extra = os.path.join(extra, 'x64')
                 assert os.path.isdir(extra), extra
-- 
2.8.0.windows.1

